lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -50,15 +54,23 @@ namespace Js
         HRESULT hr = NOERROR;
         ScriptContext* scriptContext = GetScriptContext();
         CompileScriptException se;
-        TempArenaAllocatorObject* allocatorObject = EnsureTempAllocator();
-        ArenaAllocator* allocator = allocatorObject->GetAllocator();
+        ArenaAllocator* allocator = EnsureTempAllocator();
+        *exceptionVar = nullptr;
+        if (!scriptContext->GetConfig()->IsES6ModuleEnabled())
+        {
+            return E_NOTIMPL;
+        }
         try
         {
             AUTO_NESTED_HANDLED_EXCEPTION_TYPE((ExceptionType)(ExceptionType_OutOfMemory | ExceptionType_StackOverflow));
             this->parser = (Parser*)AllocatorNew(ArenaAllocator, allocator, Parser, scriptContext);
+            this->srcInfo = {};
+            this->srcInfo.sourceContextInfo = scriptContext->CreateSourceContextInfo(sourceLength, Js::Constants::NoHostSourceContext);
+            // TODO: enable this for byte code generation.
+            // this->srcInfo.moduleID = moduleId;
```

Comment:
it's causing assert in bytecode generator as it's confused with the old moduleroot stuff. We need to have different handling for moduleId when compile with module flag on.